docs/user/cluster_setup.md

```diff
@@ -0,0 +1,282 @@
+# Running experiments on a cluster
+
+This tutorial teaches you how to setup a [Prefect](https://docs.prefect.io/) +
+[Dask distributed](https://distributed.dask.org/en/latest/) cluster to
+run multiple garage experiments in parallel on multiple machines. This setup
+requires that the absolute path to the python binary on all the machines is same
+as your primary machine in the cluster.
+
+## Installation
+
+On every machine in the cluster, install these dependencies in your virtual
+environment where garage is installed. Let's assume our virtual environment is
+called `hello-prefect`:
+
+```shell script
+source hello-prefect/bin/activate
+# we are using distributed==2.20.0 since newer versions depend on
+# cloudpickle>=1.5 which is incompatible with garage for now
+
+pip install dask distributed==2.20.0 bokeh jupyter-server-proxy prefect
+```
+
+You can find the instructions for installing garage [here](installation.html)
+
+## Bring up dask cluster and Prefect server
+
+**NOTE:** this step requires Docker + docker-compose for Prefect
+
+Select a machine to be your primary node. This machine will run dask-scheduler
+and Prefect. You can also choose to run them on different machines if you need
+to.
+
+Since the scheduler runs as a foreground process, run it in a new terminal
+session. If you are accessing your machine remotely over ssh, you can use tmux
+or similar tools to achieve this.
+
+### Start a dask distributed cluster
+
+#### Start the dask scheduler
+
+```sh
+source hello-prefect/bin/activate
+dask-scheduler  # listens on port 8786
+```
+
+Open a browser to [http://localhost:8787](http://localhost:8787) or
+`http://<scheduler machine IP>:8787`  for the Dask Scheduler UI
+
+
+#### Start dask worker
+
+On every worker machine, in a new terminal session:
+
+```sh
+source hello-prefect/bin/activate
+dask-worker <scheduler IP addr>:8786 --resources "TOKEN=1"
+```
+
+You can also run dask-worker on your primary machine.
+
+### Start prefect
+
+You can run the Prefect server and agent on the same machine as the
+dask-scheduler for simplicity.
+
+#### Set prefect config
+
+Write the following config options in a file `~/.prefect/config.toml`:
+
+```toml
+backend = "server"
+
+[server]
+host = "http://<Prefect server ip>"
+
+    [server.ui]
```

Comment:
do people usually indent toml documents?

Reply:
I used prefect's main config as base and they were using indentation. My quick google search doesn't show any strong opinions about it.